purchase_order_maintenance_equipment_link/models/maintenance_equipment.py

```diff
@@ -0,0 +1,38 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import models, fields, _
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = 'maintenance.equipment'
+
+    count_purchase_order_ids = fields.Integer(compute='_compute_purchase_orders')
+    purchase_order_ids = fields.Many2many(
+        comodel_name='purchase.order',
+        compute='_compute_purchase_orders',
+    )
+
+    def _compute_purchase_orders(self):
+        for equipment in self:
+            purchase_orders = self.env['purchase.order'].search([
+                ('order_line.maintenance_equipment_ids', 'in', equipment.id)
+            ])
```

Comment:
Add an extra condition that only enable confirmed purchases. And this condition is a bit slow, an alternative should be directly accessing order lines:
```suggestion
            purchase_orders = self.env['purchase.order.line'].search([
                ('state', 'in', ['purchase', 'done']),
                ('maintenance_equipment_ids', 'in', equipment.id),
            ]).order_id
```